docs/src/dev-guide/algorithms/task/execution.md

```diff
@@ -0,0 +1,234 @@
+5. Use APIs to retrieve outputs generated by the task from app-engine storage  
+
+## Data Provisioning
+
+Tasks receive their inputs and write their outputs as files and directories within the Task container, specifically in an input folder (e.g., `/inputs`) and an output folder (e.g., `/outputs`). The data is provisioned from Cytomine or directly through the [input provisioning API](https://github.com/cytomine/cytomine/blob/main/app-engine/src/test/resources/spec/api/openapi_spec.v0.1.0.yml) to app-engine storage,   
+
+## Task Execution
+
+App-engine schedules tasks to run in a configured kubernetes cluster, when Cytomine is deployed using docker compose it uses [k3s](https://www.rancher.com/products/k3s) to run the tasks on the other hand when it is deployed using [helm](https://github.com/cytomine/cytomine/tree/main/helm) it schedules the tasks to the same kubernetes cluster. Cytomine can be deployed on a single machine using docker compose or into a multi-node kubernetes cluster and based on these two modes of deployment the app-engine can be deployed to manage the provisioning in two ways:
+
+1. Local mode: for single machine deployment using docker compose 
+2. Cluster mode: for multi-node deplopyment
+
+### Local Mode
+
+In this mode the app-engine shares data with the task scheduled in k3s using volumes to optimize data sharing speed, the task reads and writes directly from app-engine [storage](/dev-guide/algorithms/task/#storage).
+
+To enable this mode make these entries in the docker compose yaml file in [`cytomine/compose.yml`](https://github.com/cytomine/cytomine/blob/main/compose.yaml):
+
+::: warning
+Make sure all four paths matching otherwise the data sharing fails.
+:::
+::: danger
+Since this needs absolute paths to work properly it is not automatically configured out of the box, make sure to make these changes before setting `local mode` 
+:::
+
+1. The environment variable `SCHEDULER_RUN_MODE` has two possible values `local` or `cluster`, make sure to set it to `local`.
+2. In app-engine make sure `STORAGE_BASE_PATH` and `RUN_STORAGE_BASE_PATH` both point to the absolute path of app-engine directory within cytomine data directory
+3. In app-engine define a mount bind to the absolute path of app-engine directory within cytomine data directory, make sure both host and container paths are absolute and matching.
+4. In k3s define a mount bind to the absolute path of app-engine directory within cytomine data directory, make sure both host and container paths are absolute and matching.
+
+#### Example
+
+For the app-engine:
+
+```yml
+app-engine:
+    image: cytomine/app-engine:latest
+    restart: unless-stopped
+    depends_on:
+      - postgis
+      - registry
+      - k3s
+    volumes:
+      - ${DATA_PATH:-./data}/app-engine:/data
+      - ${PWD}/.kube/shared:/root/.kube:ro
+      - /home/some-user/cytomine/data/app-engine:/home/some-user/cytomine/data/app-engine #<------ 1
```

Comment:
Instead of hardcoding the path, write something like

```suggestion
      - ${AE_DATA_PATH:-./data}/app-engine-shared-inputs:/app-engine-shared-inputs:rw
```

And then in the documentation, add that the variable `AE_DATA_PATH` should be an absolute path to the needed data and written in the `.env` file